iothub/device/src/IotHubBaseClient.cs

```diff
@@ -328,7 +341,11 @@ public async Task<long> UpdateReportedPropertiesAsync(ReportedProperties reporte
         /// This has the side-effect of subscribing to the PATCH topic on the service.
         ///  </para>
         /// </remarks>
-        /// <param name="callback">The callback to be invoked when a desired property update is received from the service.</param>
+        /// <param name="callback">
+        /// The callback to be invoked when a desired property update is received from the service.
+        /// This user-supplied callback is "fire-and-forget" and the SDK doesn't wait on it. All of requests will be processed as they arrive.
+        /// Exceptions thrown within the callback will be caught and logged by the SDK internally.
```

Comment:
If it is a fire-and-forget then the SDK won't be handling the exceptions, right? We'll need to ask users to handle exceptions within their callback implementation.